### PR TITLE
CODEOWNERS: Remove the catch-all rule

### PR DESCRIPTION
**Pull request by michi-covalent**
Remove the catch-all rule to avoid pulling in tophat for review whenever new files get added. Pulling in tophat is unnecessary because:

- lint-codeowners.yaml [^1] validates CODEOWNERS has entries for new files.
- cilium/contributing team owns CODEOWNERS file. This team can ensure that new files get placed in appropriate locations.

[^1]: https://github.com/cilium/cilium/blob/main/.github/workflows/lint-codeowners.yaml

Suggested-by: Timo Beckers <timo@isovalent.com>

### CODE REVIEW

**Comment by michi-covalent** [comment]
for the following files it might make sense to create a new `release-managers` team cc @cilium/cilium-maintainers 💭 
```
/AUTHORS
/stable.txt
/VERSION
```

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS]: nit: I don't think there's any reason @cilium/sig-policy is any more qualified for this review than @cilium/sig-agent, this is pretty generic network address code.

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS]: Yeah given the contribution history to files like this I think it makes sense to make it owned by another group responsible for releases.

**Comment by ti-mo** [comment]
Oh, TIL https://github.com/cilium/cilium/blob/main/.github/workflows/lint-codeowners.yaml. Looks like we don't really need another codeowners checker then. Thanks Michi!

**Review by ti-mo — APPROVED** [review]

File: CODEOWNERS
Comment (gandro) [inline CODEOWNERS]: I strongly feel this should be owned by @cilium/sig-lb  or `sig-datatpath`, since it's highly datapath (maglev) specific and only imported by that package. `pkg/maglev` is also owned by `sig-lb` after all

File: CODEOWNERS
Comment (tklauser) [inline CODEOWNERS]: I think this should probably be owned by a group responsible for releases.

File: CODEOWNERS
Comment (tklauser) [inline CODEOWNERS]: Same here, this should also be owned by a group responsible for releases.

File: CODEOWNERS
Comment (qmonnet) [inline CODEOWNERS:226]: Did you check whether this made any difference? From what I remember, [we ignore the catch-up rule](https://github.com/cilium/cilium/blob/main/.github/workflows/lint-codeowners.yaml#L60C1-L63) (on purpose) when validating that all files are covered in CI:

```
          # Copy all patterns from CODEOWNERS, but skipping the comments
          # ('^[^#]') and the catch-all '*' rule (the only one with a single
          # character, we skip it with '^.[^ ]').
```

**Review by nebril — APPROVED** [review]
LGTM aside from the releases team not being created yet.

File: CODEOWNERS
Comment (michi-covalent) [reply CODEOWNERS:226]: oh interesting. now i'm starting to suspect if this workflow is doing what i think it should be doing. i'll do some investigation and report back 🧑‍🔬

Comment (michi-covalent) [reply CODEOWNERS:226]: ok yeah lint-codeowners.yaml is ignoring the catch-all, so the commit message is incorrect. all these files must have added before lint-codeowners.yaml was introduced. i'll update it 📝

Comment (michi-covalent) [reply CODEOWNERS:226]: updated the commit message 📝

File: CODEOWNERS
Comment (michi-covalent) [inline CODEOWNERS:223]: i'm updating the team configuration for release-managers team. stay tuned.

https://github.com/orgs/cilium/teams/release-managers

**Review by asauber — APPROVED** [review]

**Comment by michi-covalent** [comment]
need a quick ack from @cilium/docs-structure team for the auto-generated Documentation/codeowners.rst file 🚀🙏

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS]: `.mailmap` is typically updated by @cilium/release-managers as well.

**Review by joestringer — APPROVED** [review]
Approving for codeowners. One more minor nit.

**Pull request merged**
